lib/Base/Image.cpp

```diff
@@ -415,6 +481,50 @@ bool glow::readPngImage(Tensor *T, const char *filename,
   return false;
 }
 
+bool glow::readPpmImage(Tensor *T, const char *filename,
+                        std::pair<float, float> range,
+                        llvm::ArrayRef<float> mean,
+                        llvm::ArrayRef<float> stddev) {
+  bool isGray;
+  dim_t height, width;
+  FILE *fp = fopen(filename, "rb");
+  if (!fp) {
+    return true;
+  }
+
+  // Get PPM info.
+  std::tie(height, width, isGray) = getPpmInfo(fp, filename);
+  const dim_t numChannels = isGray ? 1 : 3;
+  T->reset(ElemKind::FloatTy, {height, width, numChannels});
+
+  // Skip a single byte of space.
+  fgetc(fp);
+
+  // Read pixels and do pre-processing.
+  auto H = T->getHandle<>();
+  float scale = ((range.second - range.first) / 255.0);
+  float bias = range.first;
+  unsigned char *buf =
+      (unsigned char *)malloc(width * numChannels * sizeof(unsigned char));
```

Comment:
I'm wondering if it would be possible to start using `std::unique_ptr<>`, `std::vector<>` or `std::string` to avoid handling memory manually.